diff -uNr dpdk-18.08.org/config/common_base dpdk-18.08/config/common_base
--- dpdk-18.08.org/config/common_base	2018-08-10 05:11:26.000000000 +0800
+++ dpdk-18.08/config/common_base	2018-09-06 11:40:58.446919000 +0800
@@ -253,6 +253,38 @@
 CONFIG_RTE_LIBRTE_IXGBE_BYPASS=n
 
+#
+# Compile burst-oriented NGBE PMD driver
+#
+CONFIG_RTE_LIBRTE_NGBE_PMD=y
+CONFIG_RTE_LIBRTE_NGBE_DEBUG_RX=n
+CONFIG_RTE_LIBRTE_NGBE_DEBUG_TX=n
+CONFIG_RTE_LIBRTE_NGBE_DEBUG_TX_FREE=n
+CONFIG_RTE_NGBE_INC_VECTOR=n
+CONFIG_RTE_LIBRTE_NGBE_BYPASS=n
+CONFIG_RTE_NGBE_POLL=n
+CONFIG_RTE_NGBE_RETRY=n
+
+#
+# Compile burst-oriented TXGBE PMD driver
+#
+CONFIG_RTE_LIBRTE_TXGBE_PMD=y
+CONFIG_RTE_LIBRTE_TXGBE_DEBUG_RX=n
+CONFIG_RTE_LIBRTE_TXGBE_DEBUG_TX=n
+CONFIG_RTE_LIBRTE_TXGBE_DEBUG_TX_FREE=n
+CONFIG_RTE_TXGBE_INC_VECTOR=n
+CONFIG_RTE_LIBRTE_TXGBE_BYPASS=n
+CONFIG_RTE_TXGBE_DEBUG_BP=n
+CONFIG_RTE_TXGBE_BP_AUTO=y
+CONFIG_RTE_TXGBE_KR_POLL=n
+CONFIG_RTE_TXGBE_KR_PRESENT=n
+CONFIG_RTE_TXGBE_KX_TXRX_PIN=n
+CONFIG_RTE_TXGBE_KX_SGMII=n
+CONFIG_RTE_TXGBE_FFE_SET=n
+CONFIG_RTE_TXGBE_FFE_MAIN=0x1B1B
+CONFIG_RTE_TXGBE_FFE_PRE=0x0808
+CONFIG_RTE_TXGBE_FFE_POST=0x2C2C
+
 #
 # Compile burst-oriented I40E PMD driver
 #
 CONFIG_RTE_LIBRTE_I40E_PMD=y
diff -uNr dpdk-18.08.org/drivers/net/Makefile dpdk-18.08/drivers/net/Makefile
--- dpdk-18.08.org/drivers/net/Makefile	2018-08-10 05:11:26.000000000 +0800
+++ dpdk-18.08/drivers/net/Makefile	2018-09-06 11:41:07.422405400 +0800
@@ -29,6 +29,7 @@
 DIRS-$(CONFIG_RTE_LIBRTE_FM10K_PMD) += fm10k
 DIRS-$(CONFIG_RTE_LIBRTE_I40E_PMD) += i40e
 DIRS-$(CONFIG_RTE_LIBRTE_IXGBE_PMD) += ixgbe
+DIRS-$(CONFIG_RTE_LIBRTE_TXGBE_PMD) += netswift
 DIRS-$(CONFIG_RTE_LIBRTE_LIO_PMD) += liquidio
 DIRS-$(CONFIG_RTE_LIBRTE_MLX4_PMD) += mlx4
 DIRS-$(CONFIG_RTE_LIBRTE_MLX5_PMD) += mlx5
diff -uNr dpdk-18.08.org/MAINTAINERS dpdk-18.08/MAINTAINERS
--- dpdk-18.08.org/MAINTAINERS	2018-08-10 05:11:26.000000000 +0800
+++ dpdk-18.08/MAINTAINERS	2018-09-06 11:41:51.263936200 +0800
@@ -752,6 +752,13 @@
 F: doc/guides/nics/features/softnic.ini
 F: doc/guides/nics/softnic.rst
 
+Netswift ngbe
+M: Kenny Wang <jianwang@net-swift.com>
+T: git://dpdk.org/next/dpdk-next-netswift
+F: drivers/net/ngbe/
+F: doc/guides/nics/ngbe.rst
+F: doc/guides/nics/ngbe_vf.rst
+F: doc/guides/nics/features/ngbe*.ini
 
 Crypto Drivers
 --------------
diff -uNr dpdk-18.08.org/mk/rte.app.mk dpdk-18.08/mk/rte.app.mk
--- dpdk-18.08.org/mk/rte.app.mk	2018-08-10 05:11:26.000000000 +0800
+++ dpdk-18.08/mk/rte.app.mk	2018-09-06 11:41:23.314475200 +0800
@@ -139,6 +139,8 @@
 _LDLIBS-$(CONFIG_RTE_LIBRTE_PMD_FAILSAFE)   += -lrte_pmd_failsafe
 _LDLIBS-$(CONFIG_RTE_LIBRTE_I40E_PMD)       += -lrte_pmd_i40e
 _LDLIBS-$(CONFIG_RTE_LIBRTE_IXGBE_PMD)      += -lrte_pmd_ixgbe
+_LDLIBS-$(CONFIG_RTE_LIBRTE_NGBE_PMD)      += -lrte_pmd_ngbe
+_LDLIBS-$(CONFIG_RTE_LIBRTE_TXGBE_PMD)      += -lrte_pmd_txgbe
 ifeq ($(CONFIG_RTE_LIBRTE_KNI),y)
 _LDLIBS-$(CONFIG_RTE_LIBRTE_PMD_KNI)        += -lrte_pmd_kni
 endif
